### PR TITLE
[Issue 15151] - Check manager before refreshing object

### LINKED ISSUE #15151: Cart page throws 500 when submitting promo code with invalid session
**Sylius version affected**: 1.12.8

**Description**  
Sylius cart page throws 500 error when user session is invalid.
Issue appeared after https://github.com/Sylius/Sylius/pull/14865

**Steps to reproduce**  
Tested on Sylius 1.12.8
Also on https://demo.sylius.com/en_US/
1. Add any product to cart
2. Open browser dev tools -> Application -> Cookies 
3. Delete `PHPSESSID` cookie
4. Enter random promo code and hit submit

**Possible Solution**  
`Sylius\Bundle\OrderBundle\Controller::resetChangesOnCart action`
Change from
```
private function resetChangesOnCart(OrderInterface $cart): void
    {
        $this->manager->refresh($cart);
        foreach ($cart->getItems() as $item) {
            $this->manager->refresh($item);
        }
    }
```
To
```
    private function resetChangesOnCart(OrderInterface $cart): void
    {
        if (!$cart->getId()) {
            return;
        }

        $this->manager->refresh($cart);
        foreach ($cart->getItems() as $item) {
            $this->manager->refresh($item);
        }
    }
```

Check if cart Object is new or not before trying to refetch it from DB.

### PR DESCRIPTION
**Pull request by ulinskas**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12                  |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                      |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets | fixes #15151                       |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/OrderBundle/Controller/OrderController.php [hunk 1]
```diff
@@ -133,9 +133,11 @@ public function saveAction(Request $request): Response
...
     {
-        $this->manager->refresh($cart);
-        foreach ($cart->getItems() as $item) {
-            $this->manager->refresh($item);
+        if ($this->manager->contains($cart)) {
+            $this->manager->refresh($cart);
+            foreach($cart->getItems() as $item) {
+                $this->manager->refresh($item);
+            }
         }
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/OrderBundle/Controller/OrderController.php]:
```suggestion
        if (!$this->manager->contains($cart)) {
        	return;
        }
        
        $this->manager->refresh($cart);
        foreach($cart->getItems() as $item) {
            $this->manager->refresh($item);
        }
```

Return Early Pattern is usually more readable.

**Review by jakubtobiasz — CHANGES_REQUESTED** [review]
I asked the rest of the Core Team how we should test this change. I'll be back to this PR once we make some decision.

**Comment by jakubtobiasz** [comment]
Hello @ulinskas 👋🏼
I've added a Behat scenario covering your fix :). We'll wait for more votes, and this PR will be merged next week I guess.

**Review by jakubtobiasz — DISMISSED** [review]

File: features/cart/shopping_cart/viewing_cart_summary.feature [hunk 2]
```diff
@@ -11,3 +11,12 @@ Feature: Viewing a cart summary
     Scenario: Viewing information about empty cart
         When I see the summary of my cart
         Then my cart should be empty
+
+    @ui @no-api
+    Scenario: Viewing information about empty cart after clearing cookies
+        Given the store has a product "T-Shirt banana" priced at "$12.54"
+        And I added this product to the cart
+        And I am on the summary of my cart page
+        And my session expired
```
Comment (Zales0123) [inline features/cart/shopping_cart/viewing_cart_summary.feature]: `my session expired` is quite a technical detail :dancer: what is the observable behaviour the Customer can experience? Probably something like `But then I am gone for a long time` 😄

**Review by Zales0123 — COMMENTED** [review]
small suggestion for the scenario, but generally 👌

File: features/cart/shopping_cart/viewing_cart_summary.feature (hunk 2 above)
Comment (jakubtobiasz) [reply features/cart/shopping_cart/viewing_cart_summary.feature]: Indeed it's a technical detail, but at the same time the bug is also strictly technical issue 🤷🏼‍♂️. I'll change it.

**Review by jakubtobiasz — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @ulinskas!

**Pull request merged**